### ISSUE TITLE
Attribute error when trying to use godef

### ISSUE THREAD

**Issue by evgenybf** [ironcladlou/GoTools#101]
Win7x64, sublime 3083, go1.5

godef fails with the fallowing error (very similar to #94):
```python
reloading Packages/User/Package Control.sublime-settings
Package Control: No updated packages
Traceback (most recent call last):
  File "C:\Apps\sublime3\sublime_plugin.py", line 556, in run_
    return self.run(edit)
  File "gotools_goto_def in C:\Apps\sublime3\Data\Installed Packages\GoTools.sublime-package", line 21, in run
  File "gotools_settings in C:\Apps\sublime3\Data\Installed Packages\GoTools.sublime-package", line 19, in __init__
AttributeError: 'GoToolsSettings' object has no attribute 'GoEnv'
>>> import os
>>> os.environ['GOPATH']
'C:\\Develop\\Go\\gopath'
```

Log investigation showed that there is an initialization problem:
```python
 File "gotools_settings in C:\Apps\sublime3\Data\Installed Packages\GoTools.sublime-package", line 106, in <module>
  File "gotools_settings in C:\Apps\sublime3\Data\Installed Packages\GoTools.sublime-package", line 94, in load_goenv
  File "./subprocess.py", line 789, in __init__
  File "./subprocess.py", line 988, in _get_handles
  File "./subprocess.py", line 1038, in _make_inheritable
OSError: [WinError 6] The handle is invalid
```
https://gist.github.com/evgenybf/a3612e831cf8cf88e4e2#file-sublime-log

The line which causes the error:
```python
stdout, stderr = subprocess.Popen([gobinary, 'env'], stdout=subprocess.PIPE).communicate()
```
It's interesting that Popen does not raise the exception when I remove `stdout=subprocess.PIPE` from the argument list..

**Comment by evgenybf** [comment]
Something wrong with sublime's python:
```python
>>> import sys
>>> sys.version
'3.3.3 (v3.3.3:c3896275c0f6, Nov 18 2013, 21:19:30) [MSC v.1600 64 bit (AMD64)]'
>>> stdout, stderr = subprocess.Popen(["7z.exe"], stdout=subprocess.PIPE).communicate()
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "./subprocess.py", line 789, in __init__
  File "./subprocess.py", line 988, in _get_handles
  File "./subprocess.py", line 1038, in _make_inheritable
OSError: [WinError 6] The handle is invalid
```

The same command doesn't fall on a standalone python (2.7, 3.4 32bit)
```python
>>> sys.version
'3.4.3 (v3.4.3:9b73f1c3e601, Feb 24 2015, 22:43:06) [MSC v.1600 32 bit (Intel)]'
>>> import subprocess
>>> stdout, stderr = subprocess.Popen(["7z.exe"], stdout=subprocess.PIPE).communicate()
<no exception trace here>
```

**Comment by evgenybf** [comment]
I am very sorry for the mess here.. it turned out that the problem was on my side:
http://www.sublimetext.com/forum/viewtopic.php?f=3&t=2220#p15441
```
Re: Build system: "The handle is invalid"

Postby guillermooo on Sun Oct 02, 2011 5:09 pm
How do you start Sublime Text? Sounds like you're starting it from the command line. If so, check this out:

viewtopic.php?f=3&t=2464
```

The error doesn't occur when sublime is run from windows explorer! Usually I start it from command line with the batch script:
```batch
@C:\Apps\sublime3\subl.exe %*
```

Workaround (http://www.sublimetext.com/forum/viewtopic.php?f=3&t=2464#p11304):
```
@start C:\Apps\sublime3\sublime_text.exe %*
```

**Comment by ironcladlou** [comment]
Is the problem corrected if you change the `Popen` call to this:

```python
stdout, stderr = subprocess.Popen([gobinary, 'env'], stdout=subprocess.PIPE, stderr=subprocess.PIPE).communicate()
```

Failing to redirect stderr here can cause the target process to hang, and the Python bug you cited also seems to indicate that if output isn't properly redirected, there can be a cascade of null handling that manifests the bug.

I now correctly redirect stderr as of #99 but I haven't yet tagged a new release.

**Comment by evgenybf** [comment]
>>Is the problem corrected if you change the Popen call to this:

No, the stderr param changes nothing in the case

**Comment by ironcladlou** [comment]
I should have read the upstream bug in more detail: the error manifests if you don't redirect ALL the standard streams. If you change the code to this, does it work?

```python
p = subprocess.Popen([gobinary, 'env'], stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
p.stdin.close()
stdout, stderr = p.communicate()
```

**Comment by evgenybf** [comment]
Yep, the problem has disappeared. Both subl.exe and sublime_text.exe run from command line without any errors now.

**Comment by ironcladlou** [comment]
Great, thanks. I'll patch this.

**Comment by ironcladlou** [comment]
@evgenybf 

I merged what I think is a more succinct fix for this in #104. Would you mind either using the master branch of GoTools or manually applying the patch locally to test on your end? Thanks!

**Comment by evgenybf** [comment]
@ironcladlou, sorry, I haven't been able to check the master branch yet as GotoolsGotoDef.godef falls because of non-initialized GoToolsSettings.Instance there (for some reason it's equal to None in spite of plugin_loaded()).. But all Popen calls look correct.
